regex.c

```diff
@@ -164,79 +164,65 @@ void free_regex(void)
 	free_whitelist_domains();
```

Comment:
This function should be called separately, like what was done for `read_whitelist_from_database`.